packages/parcel-transformer-autometrics/src/lib.ts

```diff
@@ -0,0 +1,241 @@
+import path from "path";
+import {
+  EmitHint,
+  NewLineKind,
+  Node,
+  PropertyAssignment,
+  ScriptTarget,
+  TransformationContext,
+  TransformerFactory,
+  createPrinter,
+  createSourceFile,
+  isCallExpression,
+  isExpressionStatement,
+  isFunctionExpression,
+  isIdentifier,
+  isObjectLiteralExpression,
+  transform,
+  visitEachChild,
+  visitNode,
+  NodeFactory,
+  ObjectLiteralExpression,
+  isVariableDeclaration,
+} from "typescript";
+
+let moduleName: string;
+const transformerFactory: TransformerFactory<Node> = (
+  context: TransformationContext,
+) => {
+  const { factory } = context;
+  return (rootNode) => {
+    function visit(node: Node): Node {
+      node = visitEachChild(node, visit, context);
+
+      if (
+        isVariableDeclaration(node) &&
+        isIdentifier(node.name) &&
+        isCallExpression(node.initializer) &&
+        isIdentifier(node.initializer.expression) &&
+        node.initializer.expression.escapedText === "autometrics"
+      ) {
+        const [functionOrOptions, maybeFunction] = node.initializer.arguments;
+        if (isFunctionExpression(functionOrOptions)) {
+          const functionName = functionOrOptions.name.escapedText.toString();
+
+          const autometricsOptions = factory.createObjectLiteralExpression([
+            factory.createPropertyAssignment(
+              "functionName",
+              factory.createStringLiteral(functionName),
+            ),
+            factory.createPropertyAssignment(
+              "moduleName",
+              factory.createStringLiteral(moduleName),
+            ),
+          ]);
+
+          console.log(
+            `Autometrics: Adding options to ${functionName} in ${moduleName}`,
+          );
+
+          return factory.createVariableDeclaration(
+            factory.createIdentifier(node.name.escapedText.toString()),
+            undefined,
+            undefined,
+            factory.createCallExpression(
+              factory.createIdentifier("autometrics"),
+              undefined,
+              [
+                autometricsOptions,
+                factory.createFunctionExpression(
+                  undefined,
+                  undefined,
+                  factory.createIdentifier(functionName),
+                  undefined,
+                  [],
+                  undefined,
+                  factory.createBlock([], false),
+                ),
+              ],
+            ),
+          );
+        } else if (
+          isObjectLiteralExpression(functionOrOptions) &&
+          isFunctionExpression(maybeFunction)
+        ) {
+          const functionName = maybeFunction.name.escapedText.toString();
+
+          let autoMetricsOptions = addObjectPropertyIfMissing(
```

Comment:
I think we generally write autometrics as one word, right? Then it should also be a lowercase "m" here ;)

Reply:
Good catch! Updated 😄